commit cb3e1a45174b5bbbcb5279af1662b6d20c028a0d (refs/changes/95/2895/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-11-01T17:00:52+01:00 (1 year, 11 months ago)
    
    AGENT-1085 cn-agent install_agent task should not throw error trying to cleanup previous backup dirs

diff --git a/lib/tasks/agent_install.js b/lib/tasks/agent_install.js
index da61b66..bdca88a 100644
--- a/lib/tasks/agent_install.js
+++ b/lib/tasks/agent_install.js
@@ -520,7 +520,11 @@ function start() {
                     package_name + '.updating-to.' + image_uuid;
             fs.stat(backup_dir, function (er) {
                 if (er) {
-                    cb(er);
+                    if (er.code === 'ENOENT') {
+                        cb();
+                    } else {
+                        cb(er);
+                    }
                     return;
                 }
 
